design/node_gateway.md

```diff
@@ -0,0 +1,79 @@
+# Node gateway design
+
+## Scope
+
+The design for the node gateway, a component that is responsible for handling RPC requests from traditional Ethereum 
+wallets (e.g. MetaMask, hardware wallets) and webapps to the Obscuro host.
+
+## Requirements
+
+* The node gateway serves an endpoint that meets the [Ethereum JSON-RPC specification
+  ](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/ethereum/eth1.0-apis/assembled-spec/openrpc.json)
+* The node gateway is run locally by the end user
+* The node gateway does not broadcast any sensitive information (e.g. transactions, balances) in plaintext
+* The encryption keys are only known to the client (e.g. wallet, webapp), and not to any third-parties
+* The encryption is transparent to the client; from the client's perspective, they are interacting with a "standard" 
+  non-encrypting implementation of the Ethereum JSON-RPC specification
+* The node gateway is usable by any webapp or wallet type, and in particular:
+  * Hardware wallets that do not offer a decryption capability
+  * MetaMask, for which the keys are only available when running in the browser
+
+## Design
+
+The node gateway is a local server application that serves two endpoints:
+
+* An endpoint for registering new *viewing keys*
```

Comment:
for "managing" viewing keys.
They have to be:
- created
- signed by the account owner key
- eventually: revoked,  see expiration, etc